docs/src/lib/semver.md

```diff
@@ -0,0 +1,68 @@
+# Documenter & semantic versioning
+
+Documenter, [like any good Julia package](https://pkgdocs.julialang.org/v1/compatibility/#Version-specifier-format), follows [semantic versioning](https://semver.org/).
+As such, and as the package is currently in the `v1.x` era of its lifecycle, any changes in Documenter should not break any existing functionality.[^1]
+
+[^1]: Eventually, of course, Documenter `2.0` may break everything. But we don't expect a breaking release in the near future.
+
+However, Documenter is relatively complex, and it can sometimes be unclear what constitutes a _breaking change_.
+For example, is changing the CSS classes in the HTML themes allowed?
+What if the user relied on a CSS class in an `@raw html`-block?
+If their HTML is no longer rendered correctly, is that a breaking change?
+Is completely changing the LaTeX compiler a breaking change?
+What if it breaks a PDF build somewhere due to some math block using a feature that only works with `pdflatex`?
+
+This page aims to clarify what **is** and **is not** covered by the Documenter semver guarantees, both as information for users, and as guidance for developers.
+
+!!! note "This page is not complete!"
+
+    If you need to rely on something that is currently an internal API or undocumented behavior, open an issue or a pull request to get it documented!
+    There is a good chance that the behavior is already a _de factor_ semver guarantee, or can easily be cleaned up and made public API.
+    The goal is to, over time, add and document additional semver guarantees.
+
+
+## Documenter's API guarantees
+
+The following APIs and behaviors are guaranteed not to change:
+
+* Standard promises about the Julia APIs (public Documenter functions and their documented arguments).
+  In a nutshell, any `make.jl` making use of just public, documented parts of the Documenter API should always continue working (i.e. builds should complete).
```

Comment:
Could this be clarified? Does it mean all of Documenters `export`s essentially?

Reply:
I'll think about how to expand on this. It's not just exports though -- we have some unexported public stuff too, like `Documenter.HTML` etc. So it's really anything that's under the `Library > Public` section in the manual.

Reply:
But also things like the markdown syntax (`@example` etc) I guess? Perhaps it would make sense to restructure the docs a bit. Would be good to add "docstrings" to those things too and include in the API section.

Reply:
Ah, yes, should definitely reference those. And yeah, we should think about how to present them better. Not sure we strictly speaking need docstrings -- the `Manual > Syntax` page is basically the reference manual for those. But it should probably be moved under the `API` section.

Reply:
Yea, it always feels a bit strange to go to `Manual > Syntax` when looking up these.

Reply:
I slightly expanded that bullet point, and also opened #2246 to remember to reorganize the at-block page.